package/client/resource/dui/index.ts

```diff
@@ -0,0 +1,80 @@
+import {
+	GetGameTimer,
+	CreateDui,
+	GetDuiHandle,
+	CreateRuntimeTxd,
+	CreateRuntimeTextureFromDuiHandle,
+	SetDuiUrl,
+	DestroyDui,
+	SendDuiMessage,
+} from '@nativewrappers/client';
+import { cache } from '../cache';
+
+let duis: { [key: string]: Dui } = {};
+let currentId = 0;
+
+interface LibDui {
+	url: string;
+	width: number;
+	height: number;
+	debug?: boolean;
+}
+
+export class Dui {
+	id: string = "";
+	debug: boolean = false;
+	url: string = "";
+	duiObject: number = 0;
+	duiHandle: string = "";
+	runtimeTxd: number = 0;
+	txdObject: number = 0;
+	dictName: string = "";
+	txtName: string = "";
```

Comment:
Some of these should probably be marked as private to discourage manually setting the values. Same applies to the Lua implementation (I don't mind if it's real private or just typed as such).